lib/project_types/extension/loaders/project.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Extension
+  module Loaders
+    module Project
+      def self.load(context:, directory:, api_key:, registration_id:, api_secret:)
+        env_overrides = {
+          "SHOPIFY_API_KEY" => api_key,
+          "SHOPIFY_API_SECRET" => api_secret,
+          "EXTENSION_ID" => registration_id
+        }.compact
+        env = begin
+          ShopifyCLI::Resources::EnvFile.read(directory, overrides: env_overrides)
+        rescue Errno::ENOENT
+          ShopifyCLI::Resources::EnvFile.from_hash(env_overrides)
+        end
+        project = ExtensionProject.at(directory)
+        project.env = env
+        project
+      rescue SmartProperties::InitializationError
```

Comment:
I don't know if this is the correct approach to error handling, but it seemed like a `SmartProperties::InitializationError` just spits out a backtrace and reports to bugsnag, which would be undesirable.

Reply:
I'm looping it @t6d who is the expert on Smart Properties 😛